From d519df97e58fe36a81cac2c9d7b8548120a15118 Mon Sep 17 00:00:00 2001
From: Dylan Yip <dylan.yip@xilinx.com>
Date: Thu, 28 May 2020 16:46:50 -0700
Subject: [PATCH 1366/1852] v4l: xilinx: dma: Add control handler to v4l2
 device

commit e5e84d107ebd3ad55e02ad28e60c5c5eba757bdb from
https://github.com/Xilinx/linux-xlnx.git

Currently the control handler is not being passed to the v4l2 device. It
is only being held by video device. This causes a bug where the video
node is not aware of the controls implemented by its subdevices.

This patch passes the control handler to the v4l2 device.

Signed-off-by: Dylan Yip <dylan.yip@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-dma.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/xilinx/xilinx-dma.c b/drivers/media/platform/xilinx/xilinx-dma.c
index 713c865a5a6b..8a48f76528e1 100644
--- a/drivers/media/platform/xilinx/xilinx-dma.c
+++ b/drivers/media/platform/xilinx/xilinx-dma.c
@@ -1337,7 +1337,6 @@ int xvip_dma_init(struct xvip_composite_device *xdev, struct xvip_dma *dma,
 		goto error;
 	}
 
-	dma->video.ctrl_handler = &dma->ctrl_handler;
 	ret = v4l2_ctrl_handler_setup(&dma->ctrl_handler);
 	if (ret < 0) {
 		dev_err(dma->xdev->dev, "failed to set controls\n");
@@ -1347,6 +1346,7 @@ int xvip_dma_init(struct xvip_composite_device *xdev, struct xvip_dma *dma,
 	/* ... and the video node... */
 	dma->video.fops = &xvip_dma_fops;
 	dma->video.v4l2_dev = &xdev->v4l2_dev;
+	dma->video.v4l2_dev->ctrl_handler = &dma->ctrl_handler;
 	dma->video.queue = &dma->queue;
 	snprintf(dma->video.name, sizeof(dma->video.name), "%pOFn %s %u",
 		 xdev->dev->of_node,
-- 
2.31.1

